packages/design-system/src/components/popup/index.js

```diff
@@ -90,7 +90,8 @@ function Popup({
       if (evt?.target?.nodeType && popup.current?.contains(evt.target)) {
         return;
       }
-      setPopupState({
+
+      await setPopupState({
         offset:
           anchor?.current && getOffset(placement, spacing, anchor, dock, popup),
       });
```

Comment:
🤔 Does this need to be `await`ed? I don't think I've seen state setters be awaited before

Reply:
We were running to race condition issue with another `useLayoutEffect` but I think since we removed it, we probably don't need this anymore.